assignments/closure.js

```diff
@@ -7,23 +7,26 @@ const joke = [];
 
 function dadJoke() {
   joke.push("Why was 6 afraid of 7?");
-
   let setup = [6, 7, 8]
   const punchline = setup.map(x => x + 1);
   joke.push("Because", punchline);
 };
 
-dadJoke()
+dadJoke();
 console.log(joke);
 
-
 /* STRETCH PROBLEMS, Do not attempt until you have completed all previous tasks for today's project files */
 
-
+var currentCount = 0
 // ==== Challenge 2: Create a counter function ====
-const counter = () => {
+const counter = () => currentCount += 1;
   // Return a function that when invoked increments and returns a counter variable.
-};
+
```

Comment:
So this here should return a function that when called, increments a value. Take a look at this and see if you can dissect it's inner workings...
```js
const counter = (num = 0) => () => ++num;
const iter = counter();

iter(); // 1
iter(); // 2
```